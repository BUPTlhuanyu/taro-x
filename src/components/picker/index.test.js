/*
 * @Descripttion: 
 * @Author: lhuanyu
 * @Date: 2020-03-30 09:39:34
 * @LastEditors: lhuanyu
 * @LastEditTime: 2020-03-30 11:45:32
 */
import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { renderIntoDocument, Simulate } from 'nerv-test-utils'

import AtPicker from '../../../.temp/components/picker/index'

describe('Picker Snap', () => {
  it('AtPick Snap', () => {
    const component = renderToString(
      <AtPicker isOpened={true} fields='year'>
        请选择出生日期
      </AtPicker>
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Picker DOM', () => {
  it('AtPick onCancel', () => {
    const onCancel = jest.fn()
    const component = renderIntoDocument(
      <AtPicker isOpened={true} fields='year' onCancel={onCancel}>
        请选择出生日期
      </AtPicker>
    )

    const componentDom = findDOMNode(component, 'at-picker')
    const clickTargetToClose = componentDom.querySelector('.at-picker__overlay')
    
    expect(component.state._isOpened).toBeTruthy()

    Simulate.click(clickTargetToClose)
  
    process.nextTick(() => {
      expect(onCancel).toHaveBeenCalledTimes(1)
      expect(component.state._isOpened).toBeFalsy()
    })
  })
})