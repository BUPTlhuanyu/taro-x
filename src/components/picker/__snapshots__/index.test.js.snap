// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker Snap AtPick Snap 1`] = `"<div class=\\"at-picker at-picker--active\\"><div class=\\"at-picker__overlay\\"></div><div class=\\"at-picker__container picker\\"><div class=\\"picker-header\\"><div class=\\"picker-header__btn-close\\">取消</div><div class=\\"picker-header__btn\\">请选择出生日期</div><div class=\\"picker-header__btn-confirm\\">确认</div></div><div class=\\"picker-body\\"><div style=\\"flex:1;\\" class=\\"\\"><div class=\\"picker__group\\"><div class=\\"picker__group__mask\\"></div><div class=\\"picker__group__indicator\\"></div><div style=\\"padding: 1800px 0px 39000px;transform: translate3d(0, -1880px, 0);transition: transform 0s;\\" class=\\"picker__group__content\\"><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2015年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2016年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2017年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2018年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2019年</div><div style=\\"transform:scale(1.2);\\" class=\\"picker__group__content__item\\">2020年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2021年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2022年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2023年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2024年</div><div style=\\"transform:scale(1);\\" class=\\"picker__group__content__item\\">2025年</div></div></div></div>undefinedundefined</div></div></div>"`;
